### PR TITLE
[WFCORE-4635] Introducing new CapabilityServiceBuilder.provides() method

### PR DESCRIPTION
**Pull request by ropalka**
https://issues.jboss.org/browse/WFCORE-4635

### CODE REVIEW

File: controller/src/main/java/org/jboss/as/controller/CapabilityServiceBuilder.java [hunk 1]
```diff
@@ -114,11 +114,41 @@
     /**
      * Provided capability.
      *
-     * @param capabilities names of capabilities service will provide.
+     * @param capability name of capability provided value represent
      * @param <V> consumed value type
      * @return consumer providing value
      */
-    <V> Consumer<V> provides(final RuntimeCapability<?>... capabilities);
+    <V> Consumer<V> provides(final RuntimeCapability<?> capability);
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/CapabilityServiceBuilder.java]: This is a breaking API change.

Comment (ropalka) [reply controller/src/main/java/org/jboss/as/controller/CapabilityServiceBuilder.java]: Thanks Brian for catching it up!

**Comment by ropalka** [comment]
Breaking API change have been fixed @bstansberry

**Comment by ropalka** [comment]
How about this few months pending PR @bstansberry  ?
Any chance it will get it into upstream anytime soon?
I have some pending work depending on it.

**Comment by jmesnil** [comment]
@ropalka thanks

**Pull request merged**